src/libImaging/Fill.c

```diff
@@ -76,8 +76,21 @@ ImagingFillLinearGradient(const char *mode) {
         return NULL;
     }
 
-    for (y = 0; y < 256; y++) {
-        memset(im->image8[y], (unsigned char)y, 256);
+    if (im->image8) {
+        for (y = 0; y < 256; y++) {
+            memset(im->image8[y], (unsigned char)y, 256);
+        }
+    } else {
+        int x;
+        for (y = 0; y < 256; y++) {
+            for (x = 0; x < 256; x++) {
+                if (im->type == IMAGING_TYPE_FLOAT32) {
```

Comment:
Does refactoring this check outside the `256 * 256` loop make any sort of performance difference?

Reply:
Branch prediction should be bang on for this one. And I'd even expect that the compiler would hoist this.